// @redis-ai/langcache/dist/esm/sdk/sdk.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { conversationalSearch } from "../funcs/conversationalSearch.js";
import { deleteById } from "../funcs/deleteById.js";
import { deleteQuery } from "../funcs/deleteQuery.js";
import { flush } from "../funcs/flush.js";
import { search } from "../funcs/search.js";
import { set } from "../funcs/set.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class LangCache extends ClientSDK {
    /**
     * Searches the cache for entries using conversational context. When conversation context is provided, the query is reformulated before searching.
     */
    async conversationalSearch(request, options) {
        return unwrapAsync(conversationalSearch(this, request, options));
    }
    /**
     * Deletes multiple cache entries based on specified attributes. If no attributes are provided, all entries in the cache are deleted.
     */
    async deleteQuery(request, options) {
        return unwrapAsync(deleteQuery(this, request, options));
    }
    /**
     * Adds an entry to the cache with a prompt and response.
     */
    async set(request, options) {
        return unwrapAsync(set(this, request, options));
    }
    /**
     * Searches the cache for entries that match the prompt and attributes. If no entries are found, this endpoint returns an empty array.
     */
    async search(request, options) {
        return unwrapAsync(search(this, request, options));
    }
    /**
     * Deletes a single cache entry by the entry ID.
     */
    async deleteById(entryId, options) {
        return unwrapAsync(deleteById(this, entryId, options));
    }
    /**
     * Flushes all entries from the cache.
     */
    async flush(request, options) {
        return unwrapAsync(flush(this, request, options));
    }
}
