// @redis-ai/langcache/src/models/cacheentry.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./errors/sdkvalidationerror.js";
import {
  SearchStrategy,
  SearchStrategy$inboundSchema,
} from "./searchstrategy.js";

export type CacheEntry = {
  /**
   * Unique ID for the cache entry.
   */
  id: string;
  /**
   * The prompt associated with the cache entry.
   */
  prompt: string;
  /**
   * The response associated with the cache entry.
   */
  response: string;
  attributes: { [k: string]: string };
  /**
   * The similarity metric used for similarity comparison.
   */
  similarity: number;
  searchStrategy: SearchStrategy;
};

/** @internal */
export const CacheEntry$inboundSchema: z.ZodType<
  CacheEntry,
  z.ZodTypeDef,
  unknown
> = z.object({
  id: z.string(),
  prompt: z.string(),
  response: z.string(),
  attributes: z.record(z.string()),
  similarity: z.number(),
  searchStrategy: SearchStrategy$inboundSchema,
});

export function cacheEntryFromJSON(
  jsonString: string,
): SafeParseResult<CacheEntry, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => CacheEntry$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'CacheEntry' from JSON`,
  );
}
