// @redis-ai/langcache/src/models/conversationalsearchresponsecontent.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { CacheEntry, CacheEntry$inboundSchema } from "./cacheentry.js";
import { SDKValidationError } from "./errors/sdkvalidationerror.js";

/**
 * Response representing the result of a conversational cache search operation
 */
export type ConversationalSearchResponseContent = {
  /**
   * The actual prompt used for vector search. When CQR reformulates the query, this contains the reformulated version. Otherwise equals the original prompt.
   */
  actualPrompt: string;
  /**
   * Array of cache entries matching the search criteria. This array is empty if no entries match the search criteria.
   */
  data: Array<CacheEntry>;
};

/** @internal */
export const ConversationalSearchResponseContent$inboundSchema: z.ZodType<
  ConversationalSearchResponseContent,
  z.ZodTypeDef,
  unknown
> = z.object({
  actualPrompt: z.string(),
  data: z.array(CacheEntry$inboundSchema),
});

export function conversationalSearchResponseContentFromJSON(
  jsonString: string,
): SafeParseResult<ConversationalSearchResponseContent, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) =>
      ConversationalSearchResponseContent$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ConversationalSearchResponseContent' from JSON`,
  );
}
