// @redis-ai/langcache/src/models/conversationalsearchbody.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import {
  SearchStrategy,
  SearchStrategy$outboundSchema,
} from "./searchstrategy.js";

export type ConversationalSearchBody = {
  /**
   * The prompt to search for in the cache.
   */
  prompt: string;
  /**
   * The minimum similarity threshold for the cache entry (normalized cosine similarity).
   */
  similarityThreshold?: number | undefined;
  /**
   * The search strategies to use for the search, ordered by priority.
   */
  searchStrategies?: Array<SearchStrategy> | undefined;
  attributes?: { [k: string]: string } | undefined;
  /**
   * The maximum number of results to return. Defaults to 1 if not specified.
   */
  maxResults?: number | undefined;
  /**
   * Conversation history in chronological order [Q1, Q2, Q3, ...]. When provided, the query is reformulated using CQR before searching.
   */
  context?: Array<string> | undefined;
};

/** @internal */
export type ConversationalSearchBody$Outbound = {
  prompt: string;
  similarityThreshold?: number | undefined;
  searchStrategies?: Array<string> | undefined;
  attributes?: { [k: string]: string } | undefined;
  maxResults?: number | undefined;
  context?: Array<string> | undefined;
};

/** @internal */
export const ConversationalSearchBody$outboundSchema: z.ZodType<
  ConversationalSearchBody$Outbound,
  z.ZodTypeDef,
  ConversationalSearchBody
> = z.object({
  prompt: z.string(),
  similarityThreshold: z.number().optional(),
  searchStrategies: z.array(SearchStrategy$outboundSchema).optional(),
  attributes: z.record(z.string()).optional(),
  maxResults: z.number().int().optional(),
  context: z.array(z.string()).optional(),
});

export function conversationalSearchBodyToJSON(
  conversationalSearchBody: ConversationalSearchBody,
): string {
  return JSON.stringify(
    ConversationalSearchBody$outboundSchema.parse(conversationalSearchBody),
  );
}
