// @redis-ai/langcache/src/models/fielderror.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { Result as SafeParseResult } from "../types/fp.js";
import { SDKValidationError } from "./errors/sdkvalidationerror.js";

/**
 * Validation error details for a single request field.
 */
export type FieldError = {
  /**
   * Name of the invalid request field.
   */
  field: string;
  /**
   * Validation rule that was violated.
   */
  rule: string;
  /**
   * Human-readable validation error message.
   */
  message: string;
};

/** @internal */
export const FieldError$inboundSchema: z.ZodType<
  FieldError,
  z.ZodTypeDef,
  unknown
> = z.object({
  field: z.string(),
  rule: z.string(),
  message: z.string(),
});

export function fieldErrorFromJSON(
  jsonString: string,
): SafeParseResult<FieldError, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => FieldError$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'FieldError' from JSON`,
  );
}
