// @redis-ai/langcache/src/models/searchrequest.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import {
  SearchStrategy,
  SearchStrategy$outboundSchema,
} from "./searchstrategy.js";

export type SearchRequest = {
  /**
   * The prompt to search for in the cache.
   */
  prompt: string;
  /**
   * The minimum similarity threshold for the cache entry (normalized cosine similarity).
   */
  similarityThreshold?: number | undefined;
  /**
   * The search strategies to use for the search, ordered by priority.
   */
  searchStrategies?: Array<SearchStrategy> | undefined;
  attributes?: { [k: string]: string } | undefined;
  /**
   * The maximum number of results to return. Defaults to 1 if not specified.
   */
  maxResults?: number | undefined;
};

/** @internal */
export type SearchRequest$Outbound = {
  prompt: string;
  similarityThreshold?: number | undefined;
  searchStrategies?: Array<string> | undefined;
  attributes?: { [k: string]: string } | undefined;
  maxResults?: number | undefined;
};

/** @internal */
export const SearchRequest$outboundSchema: z.ZodType<
  SearchRequest$Outbound,
  z.ZodTypeDef,
  SearchRequest
> = z.object({
  prompt: z.string(),
  similarityThreshold: z.number().optional(),
  searchStrategies: z.array(SearchStrategy$outboundSchema).optional(),
  attributes: z.record(z.string()).optional(),
  maxResults: z.number().int().optional(),
});

export function searchRequestToJSON(searchRequest: SearchRequest): string {
  return JSON.stringify(SearchRequest$outboundSchema.parse(searchRequest));
}
