// @redis-ai/langcache/src/models/setrequest.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";

export type SetRequest = {
  /**
   * The prompt for the entry.
   */
  prompt: string;
  /**
   * The response to the prompt for the entry.
   */
  response: string;
  attributes?: { [k: string]: string } | undefined;
  /**
   * The entry's time-to-live, in milliseconds. If not set, the cache's default TTL is used.
   */
  ttlMillis?: number | undefined;
};

/** @internal */
export type SetRequest$Outbound = {
  prompt: string;
  response: string;
  attributes?: { [k: string]: string } | undefined;
  ttlMillis?: number | undefined;
};

/** @internal */
export const SetRequest$outboundSchema: z.ZodType<
  SetRequest$Outbound,
  z.ZodTypeDef,
  SetRequest
> = z.object({
  prompt: z.string(),
  response: z.string(),
  attributes: z.record(z.string()).optional(),
  ttlMillis: z.number().int().optional(),
});

export function setRequestToJSON(setRequest: SetRequest): string {
  return JSON.stringify(SetRequest$outboundSchema.parse(setRequest));
}
