// @redis-ai/langcache/dist/esm/models/errors/authenticationerrorresponsecontent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import * as models from "../index.js";
import { LangCacheError } from "./langcacheerror.js";
/**
 * Authentication credentials are missing, malformed, or invalid.
 */
export class AuthenticationErrorResponseContent extends LangCacheError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.title = err.title;
        if (err.status != null)
            this.status = err.status;
        if (err.detail != null)
            this.detail = err.detail;
        if (err.instance != null)
            this.instance = err.instance;
        this.type = err.type;
        this.name = "AuthenticationErrorResponseContent";
    }
}
/** @internal */
export const AuthenticationErrorResponseContent$inboundSchema = z.object({
    title: z.string(),
    status: z.number().int().default(401),
    detail: z.string().optional(),
    instance: z.string().optional(),
    type: models.AuthenticationErrorType$inboundSchema,
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new AuthenticationErrorResponseContent(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
