// @redis-ai/langcache/src/lib/config.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { HTTPClient } from "./http.js";
import { Logger } from "./logger.js";
import { RetryConfig } from "./retries.js";
import { pathToFunc } from "./url.js";

export type SDKOptions = {
  apiKey?: string | (() => Promise<string>) | undefined;

  /**
   * Allows setting the cacheId parameter for all supported operations
   */
  cacheId?: string | undefined;

  httpClient?: HTTPClient;
  /**
   * Specifies the server URL to be used by the SDK
   */
  serverURL: string;
  /**
   * Allows overriding the default user agent used by the SDK
   */
  userAgent?: string | undefined;
  /**
   * Allows overriding the default retry config used by the SDK
   */
  retryConfig?: RetryConfig;
  timeoutMs?: number;
  debugLogger?: Logger;
};

export function serverURLFromOptions(options: SDKOptions): URL | null {
  const serverURL = options.serverURL;

  if (!serverURL) {
    return null;
  }

  const params: Record<string, string | undefined> = {};

  const u = pathToFunc(serverURL)(params);
  return new URL(u);
}

export const SDK_METADATA = {
  language: "typescript",
  sdkVersion: "0.12.0",
  genVersion: "2.907.0",
  userAgent: "speakeasy-sdk/typescript 0.12.0 2.907.0 @redis-ai/langcache",
} as const;
