// @redis-ai/langcache/dist/commonjs/models/errors/badrequesterrorresponsecontent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BadRequestErrorResponseContent$inboundSchema = exports.BadRequestErrorResponseContent = void 0;
const z = __importStar(require("zod/v3"));
const models = __importStar(require("../index.js"));
const langcacheerror_js_1 = require("./langcacheerror.js");
/**
 * Request validation or input decoding failed.
 */
class BadRequestErrorResponseContent extends langcacheerror_js_1.LangCacheError {
    constructor(err, httpMeta) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        this.title = err.title;
        if (err.status != null)
            this.status = err.status;
        if (err.detail != null)
            this.detail = err.detail;
        if (err.instance != null)
            this.instance = err.instance;
        this.type = err.type;
        if (err.fields != null)
            this.fields = err.fields;
        this.name = "BadRequestErrorResponseContent";
    }
}
exports.BadRequestErrorResponseContent = BadRequestErrorResponseContent;
/** @internal */
exports.BadRequestErrorResponseContent$inboundSchema = z.object({
    title: z.string(),
    status: z.number().int().default(400),
    detail: z.string().optional(),
    instance: z.string().optional(),
    type: models.BadRequestErrorType$inboundSchema,
    fields: z.array(models.FieldError$inboundSchema).optional(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new BadRequestErrorResponseContent(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
