// @redis-ai/langcache/dist/commonjs/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LangCache = void 0;
const conversationalSearch_js_1 = require("../funcs/conversationalSearch.js");
const deleteById_js_1 = require("../funcs/deleteById.js");
const deleteQuery_js_1 = require("../funcs/deleteQuery.js");
const flush_js_1 = require("../funcs/flush.js");
const search_js_1 = require("../funcs/search.js");
const set_js_1 = require("../funcs/set.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class LangCache extends sdks_js_1.ClientSDK {
    /**
     * Searches the cache for entries using conversational context. When conversation context is provided, the query is reformulated before searching.
     */
    async conversationalSearch(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, conversationalSearch_js_1.conversationalSearch)(this, request, options));
    }
    /**
     * Deletes multiple cache entries based on specified attributes. If no attributes are provided, all entries in the cache are deleted.
     */
    async deleteQuery(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, deleteQuery_js_1.deleteQuery)(this, request, options));
    }
    /**
     * Adds an entry to the cache with a prompt and response.
     */
    async set(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, set_js_1.set)(this, request, options));
    }
    /**
     * Searches the cache for entries that match the prompt and attributes. If no entries are found, this endpoint returns an empty array.
     */
    async search(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, search_js_1.search)(this, request, options));
    }
    /**
     * Deletes a single cache entry by the entry ID.
     */
    async deleteById(entryId, options) {
        return (0, fp_js_1.unwrapAsync)((0, deleteById_js_1.deleteById)(this, entryId, options));
    }
    /**
     * Flushes all entries from the cache.
     */
    async flush(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, flush_js_1.flush)(this, request, options));
    }
}
exports.LangCache = LangCache;
