// @redis-ai/langcache/src/models/errors/payloadtoolargeerrorresponsecontent.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod/v3";
import * as models from "../index.js";
import { LangCacheError } from "./langcacheerror.js";

/**
 * The request payload exceeds the maximum supported size.
 */
export type PayloadTooLargeErrorResponseContentData = {
  /**
   * A short, human-readable summary of the problem
   *
   * @remarks
   *                                       type.  It SHOULD NOT change from occurrence to occurrence of the
   *                                       problem, except for purposes of localization (e.g., using
   *                                       proactive content negotiation; see [RFC7231], Section 3.4).
   */
  title: string;
  /**
   * The HTTP status code ([RFC7231], Section 6) generated by the origin server for this occurrence of the problem.
   */
  status?: number | undefined;
  /**
   * A human-readable explanation specific to this occurrence of the problem.
   */
  detail?: string | undefined;
  /**
   * A URI reference that identifies the specific occurrence of the problem.  It may or may not yield further information if dereferenced.
   */
  instance?: string | undefined;
  /**
   * Problem type URI for payload-too-large errors.
   */
  type: models.PayloadTooLargeErrorType;
};

/**
 * The request payload exceeds the maximum supported size.
 */
export class PayloadTooLargeErrorResponseContent extends LangCacheError {
  /**
   * A short, human-readable summary of the problem
   *
   * @remarks
   *                                       type.  It SHOULD NOT change from occurrence to occurrence of the
   *                                       problem, except for purposes of localization (e.g., using
   *                                       proactive content negotiation; see [RFC7231], Section 3.4).
   */
  title: string;
  /**
   * The HTTP status code ([RFC7231], Section 6) generated by the origin server for this occurrence of the problem.
   */
  status?: number | undefined;
  /**
   * A human-readable explanation specific to this occurrence of the problem.
   */
  detail?: string | undefined;
  /**
   * A URI reference that identifies the specific occurrence of the problem.  It may or may not yield further information if dereferenced.
   */
  instance?: string | undefined;
  /**
   * Problem type URI for payload-too-large errors.
   */
  type: models.PayloadTooLargeErrorType;

  /** The original data that was passed to this error instance. */
  data$: PayloadTooLargeErrorResponseContentData;

  constructor(
    err: PayloadTooLargeErrorResponseContentData,
    httpMeta: { response: Response; request: Request; body: string },
  ) {
    const message = "message" in err && typeof err.message === "string"
      ? err.message
      : `API error occurred: ${JSON.stringify(err)}`;
    super(message, httpMeta);
    this.data$ = err;
    this.title = err.title;
    if (err.status != null) this.status = err.status;
    if (err.detail != null) this.detail = err.detail;
    if (err.instance != null) this.instance = err.instance;
    this.type = err.type;

    this.name = "PayloadTooLargeErrorResponseContent";
  }
}

/** @internal */
export const PayloadTooLargeErrorResponseContent$inboundSchema: z.ZodType<
  PayloadTooLargeErrorResponseContent,
  z.ZodTypeDef,
  unknown
> = z.object({
  title: z.string(),
  status: z.number().int().default(413),
  detail: z.string().optional(),
  instance: z.string().optional(),
  type: models.PayloadTooLargeErrorType$inboundSchema,
  request$: z.instanceof(Request),
  response$: z.instanceof(Response),
  body$: z.string(),
})
  .transform((v) => {
    return new PayloadTooLargeErrorResponseContent(v, {
      request: v.request$,
      response: v.response$,
      body: v.body$,
    });
  });
